Source/qol/itemlabels.cpp

```diff
@@ -0,0 +1,170 @@
+/**
+* @file itemlabels.cpp
+*
+* Adds Item Labels QoL feature
+*/
+
+#include "common.h"
+#include "control.h"
+#include "itemlabels.h"
+#include "inv.h"
+#include "render.h"
+#include "gmenu.h"
+#include "cursor.h"
+
+#include <unordered_set>
+#include <vector>
+#include <string>
+
+namespace devilution {
+
+namespace {
+
+struct itemLabelQueue {
+	int itemID, row, col, x, y, width, height;
+	text_color color;
+	std::string text;
+};
+
+std::vector<itemLabelQueue> labelQ;
+int drawMinX;
+int drawMaxX;
+bool altPressed = false;
+int generatingLabelsState = 0; // 0 - waiting for the game to start, 1 = started generating, 2 = finished generating
```

Comment:
Can use named states instead of comments:

```suggestion
enum class State {
  NOT_STARTED = 0,
  STARTED,
  FINISHED,
};

State generatingLabelsState;
```